vm/actor/src/builtin/market/mod.rs

```diff
@@ -4,7 +4,710 @@
+        let mut total_deal_space_time = BigInt::zero();
+        let mut deal_weight = BigInt::zero();
+
+        rt.transaction::<State, Result<(), ActorError>, _>(|st, rt| {
+            // if there are no dealIDs, it is a CommittedCapacity sector
+            // and the totalDealSpaceTime should be zero
+            let mut states = Amt::load(&st.states, rt.store())
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+            let proposals = Amt::load(&st.proposals, rt.store())
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            for id in &params.deal_ids {
+                let mut deal: DealState = states
+                    .get(*id)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?
+                    .ok_or_else(|| {
+                        ActorError::new(
+                            ExitCode::ErrIllegalState,
+                            "Failed to retrieve the DealState".to_owned(),
+                        )
+                    })?;
+                let proposal: DealProposal = proposals
+                    .get(*id)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?
+                    .ok_or_else(|| {
+                        ActorError::new(
+                            ExitCode::ErrIllegalState,
+                            "Failed to retrieve the DealProposal".to_owned(),
+                        )
+                    })?;
+
+                validate_deal_can_activate(
+                    rt.curr_epoch(),
+                    &miner_addr,
+                    params.sector_expiry,
+                    &deal,
+                    &proposal,
+                )?;
+
+                deal.sector_start_epoch = OptionalEpoch(Some(rt.curr_epoch()));
+                states
+                    .set(*id, deal)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+                // compute deal weight
+                let deal_space_time = proposal.duration() * proposal.piece_size.0;
+                total_deal_space_time += deal_space_time;
+            }
+            st.states = states
+                .flush()
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            let epoch_value = params
+                .sector_expiry
+                .checked_sub(rt.curr_epoch())
+                .ok_or_else(|| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalArgument,
+                        format!(
+                            "invalid sector expiry epoch {:?} ahead of current {:?}",
+                            params.sector_expiry,
+                            rt.curr_epoch()
+                        ),
+                    )
+                })?;
+            let sector_space_time: BigInt = BigInt::from(params.sector_size as u64 * epoch_value);
```

Comment:
You need to wrap the `params.sector_size` in a big int before doing the multiplication, this did not prevent the overflow I mentioned before